.github/workflows/ci.yml

```diff
@@ -88,12 +88,12 @@ jobs:
         include:
           - package_name: asdf-standard
             repository: asdf-format/asdf-standard
-            ref: master
+            ref: main
```

Comment:
These can be deleted rather than changed to `main`.

Reply:
False, more changes than just removing the ref are required. The workflow later references this.

I'd rather minimize the changes to what is actually the topic of this PR, namely backporting a pin for jsonschema, rather than refactoring a bunch of unrelated things. The main goal with the other changes is to simply get the CI working well enough to establish that this doesn't break ASDF.

Reply:
True... you are :)

Leaving out those changes sounds good. The devdeps errors look to be what was solved in https://github.com/asdf-format/asdf/pull/1470